api/graphite_proxy.go

```diff
@@ -0,0 +1,50 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"sync"
+
+	"github.com/raintank/metrictank/stats"
+)
+
+var proxyStats graphiteProxyStats
+
+func init() {
+	proxyStats = graphiteProxyStats{
+		funcMiss: make(map[string]*stats.Counter32),
+	}
+
+}
+
+type graphiteProxyStats struct {
+	sync.Mutex
+	funcMiss map[string]*stats.Counter32
+}
+
+func (s *graphiteProxyStats) Miss(fun string) {
+	s.Lock()
+	counter, ok := s.funcMiss[fun]
+	if !ok {
+		counter = stats.NewCounter32(fmt.Sprintf("api.request.render.proxy-due-to.%s", fun))
+		s.funcMiss[fun] = counter
+	}
+	s.Unlock()
+	counter.Inc()
+}
+
+func NewGraphiteProxy(u *url.URL) *httputil.ReverseProxy {
+	graphiteProxy := httputil.NewSingleHostReverseProxy(u)
+	// remove these headers from upstream. we will set our own correct ones
+	graphiteProxy.ModifyResponse = func(resp *http.Response) error {
+		// if kept, would be duplicated. and duplicated headers are illegal)
+		resp.Header.Del("access-control-allow-credentials")
+		resp.Header.Del("Access-Control-Allow-Origin")
```

Comment:
it looks a little confusing that the header on `:43` is with lower case and `:44` is with uppercase letters. `.Del()` should be case insensitive anyway